diesel/src/associations/belongs_to.rs

```diff
@@ -5,10 +5,14 @@ use helper_types::{FindBy, Filter};
 use prelude::*;
 use super::{Identifiable, HasTable};
 
-pub trait BelongsTo<Parent: Identifiable> {
+use std::borrow::Borrow;
+use std::hash::Hash;
+
+pub trait BelongsTo<Parent> {
+    type ForeignKey: Hash + ::std::cmp::Eq;
     type ForeignKeyColumn: Column;
 
-    fn foreign_key(&self) -> Option<&Parent::Id>;
+    fn foreign_key(&self) -> Option<&Self::ForeignKey>;
```

Comment:
These two additions make so much sense. It's much clearer how this is used in grouped_by now.